coc/war_clans.py

```diff
@@ -147,6 +147,15 @@ def defenses(self) -> typing.List["WarAttack"]:
         """
         other = self._war and self._war.clan if self.is_opponent else self._war.opponent
         return other.attacks if other else []
+    
+    @property
+    def average_attack_duration(self) -> int:
+        """:class:`int`: Returns the average duration of all clan member's attacks this war."""
+        total_duration = 0
+        count = len(self._cs_attacks)
```

Comment:
The way `_cs_attacks` works is that it caches (or saves) the attacks in that attribute after calculating it when `WarClan.attacks` is called for the first time, to save having to get all the attacks every time. This has the downside that `_cs_attacks` will be empty (more specifically, not set) if `WarClan.attacks` hasn't been called.

It's probably best to just use `self.attacks` here 😃